jobrunner/executors/local.py

```diff
@@ -327,20 +362,23 @@ def finalize_job(job_definition):
     message = None
 
     # special case OOMKilled
+    # Nb. this flag has been observed to be unreliable on some versions of Linux
     if container_metadata["State"]["OOMKilled"]:
         message = "Ran out of memory"
         memory_limit = container_metadata.get("HostConfig", {}).get("Memory", 0)
         if memory_limit > 0:
             gb_limit = memory_limit / (1024**3)
             message += f" (limit for this job was {gb_limit:.2f}GB)"
+    elif exit_code == 137 and job_definition.cancelled:
```

Comment:
Hmmm, I had (completely evidence-free) assumed:
* that the OOMKilled flag was giving false negatives - i.e. if it's set then it was definitely OOM, if it's not set then it might be OOM or it might be something else
*  if cancel was set, the job would be killed fairly expediently - so hardly any jobs would be cancelled & then go on to hitting OOM before actually exiting
so I was therefore thinking of these two branches as being fairly orthogonal & it doesn't matter which way round they go. But if you think otherwise, then we can change them around. 
The interesting case is if OOMKilled is false negative & we fall through to returning "Killed by an OpenSAFELY admin" (or "Job Killed" as this PR currently has it)

Reply:
Yeah, that's fair.

I guess we could have a race condition where a job's marked as cancelled, but before we get to it, its OOMKilled, in which case this might misreport (sort of!)

I have a mild preference for handling the trusted explicit info (cancelled) first, but as you say, not a big issue.

Reply:
cool, re-ordered